CodeU_Assignement_4/src/codeu_assignement_4/KEMIslandsFinder.java

```diff
@@ -0,0 +1,127 @@
+package codeu_assignement_4;
+
+/**
+ *
+ * @author Karine
+ * 
+ * The class is an Algorithm that starts from 
+ * the only public method countIslands which 
+ * gets: a map and the size of the map (rxc)
+ * and returns the number of Islands in the 
+ * map
+ */
+public class KEMIslandsFinder {
+    
+    private boolean[][] m_visisted_map;
+    
+    public int countIslands(int row, int col, boolean[][] map)
+    {        
+        if ((row > 0) && (col > 0))
+        {
+            if ((map.length == row) && (map[0].length == col))
+            {
+                m_visisted_map = new boolean[row][col];
+                initVisitedMap();
+                return this.countIslands(map, 0, 0);
+            } 
+            else
+            {
+                throw new IllegalArgumentException();
+            }
+        }
+        
+        return 0;
+    }
+    
+    /* Private Helpers */
+    private int countIslands(boolean[][] map, int curr_row, int curr_col)

```

Comment:
This method could be written a lot more simply as a pair of nested for loops iterating over the rows and columns.